src/CTA.WebForms/TagCodeBehindHandlers/TagCodeBehindHandler.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CTA.WebForms.TagCodeBehindHandlers
+{
+    public abstract class TagCodeBehindHandler
+    {
+        /// <summary>
+        /// The expected type representation in the code behind for the control that
+        /// this handler is tasked with handling.
+        /// </summary>
+        public string CodeBehindType { get; }
+        /// <summary>
+        /// The value of the ID attribute for the control that this handler is tasked with
+        /// handling.
+        /// </summary>
+        public string IdValue { get; }
+        /// <summary>
+        /// Conversions that have been staged for code behind references to the control
+        /// that this handler is tasked with handling.
+        /// </summary>
+        public IDictionary<SyntaxNode, SyntaxNode> StagedConversions;
```

Comment:
this seems little bit weird, if key and value both are syntaxNode why  not to use a hashset

Reply:
Hash set is a better idea, will change